Update sl-jdbc.md (#4604)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud-apis/sl-jdbc.md

Where Filters have a few objects that you can use:

> - `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`

Reviewer: ```suggestion
- `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`.
```

> - `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`

Reviewer: ```suggestion
- `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`.
```

> - `Entity()` &mdash;  Used for entities like primary and foreign keys - `Entity('order_id')`

Reviewer: ```suggestion
- `Entity()` &mdash;  Used for entities like primary and foreign keys - `Entity('order_id')`.
```

Where Filters have a few objects that you can use:

> - `Dimension()` - Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`

Reviewer 2: ```suggestion
- `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
```

> - `TimeDimension()` - Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`

Reviewer 2: ```suggestion
- `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`
```

> - `Entity()` - Used for entities like primary and foreign keys - `Entity('order_id')`

Reviewer 2: ```suggestion
- `Entity()` &mdash;  Used for entities like primary and foreign keys - `Entity('order_id')`
```

> For `TimeDimension()`, the grain is only required in the WHERE filter if the aggregation time dimensions for the measures & metrics associated with the where filter have different grains. 

Reviewer 2: ```suggestion
For `TimeDimension()`, the grain is only required in the `WHERE` filter if the aggregation time dimensions for the measures and metrics associated with the where filter have different grains. 
```

> Note: If you prefer a more explicit path to create the `where` clause, you can optionally use the `TimeDimension` feature. This helps separate out categorical dimensions from time-related ones. The `TimeDimension` input takes the time dimension name optionally requires granularity, like this: `TimeDimension('metric_time', 'month')`.

Reviewer 3: Maybe `explicit path` -> `strongly typed` since that phrase was used before?

> For example, consider this Semantic Model and Metric Config which contains two metrics that are aggregated across different time grains. This example shows a single Semantic Model, but the same goes for metrics across more than one Semantic Model.

Reviewer 2: ```suggestion
For example, consider this Semantic model and Metric config, which contains two metrics that are aggregated across different time grains. This example shows a single semantic model, but the same goes for metrics across more than one semantic model.
```

> For both `TimeDimension()` and `Dimension()` objects, the grain is only required if in the WHERE filter if the aggregation time dimensions for the measures & metrics associated with the where filter have different grains. 

Reviewer 4: Grain is still required when querying a time dimension using `Dimension()`. For example you would still need to add `Dimensions('metric_time','day')`.

For example, consider this Semantic model and Metric config, which contains two metrics that are aggregated across different time grains. This example shows a single semantic model, but the same goes for metrics across more than one semantic model.

```yaml
semantic_model:
  name: my_model_source

defaults:
  agg_time_dimension: created_month
  measures:
    - name: measure_0
      agg: sum
    - name: measure_1
      agg: sum
      agg_time_dimension: order_year
  dimensions:
    - name: created_month
      type: time
      type_params:
        time_granularity: month
    - name: order_year
      type: time
      type_params:
        time_granularity: year

metrics:
  - name: metric_0
    description: A metric with a month grain.
    type: simple
    type_params:
      measure: measure_0
  - name: metric_1
    description: A metric with a year grain.
    type: simple
    type_params:
      measure: measure_1

```

Assuming the user is querying `metric_0` and `metric_1` together in a single request, a valid `WHERE` filter would be:

  * `"{{ TimeDimension('metric_time', 'year') }} > '2020-01-01'"`

> Invalid Filters would be:

Reviewer 2: ```suggestion
Invalid filters would be:
```

  limit=10,
  order_by=[-'order_gross_profit']
  }}
```

> If you are ordering by an object that's been operated on (e.g., you changed the the granularity of the time dimension), or you are using the full object notation, descending order must look like:

Reviewer 2: ```suggestion
If you are ordering by an object that's been operated on (for example, you changed the the granularity of the time dimension), or you are using the full object notation, descending order must look like:
```

## FAQs

> <FAQ path="Troubleshooting/sl-alpn-error" /><br />

Reviewer 2: ```suggestion
<FAQ path="Troubleshooting/sl-alpn-error" />
```
